File: app/src/main/java/com/otraempresa/android/example/AbstractRequestSecureValidationInfoAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.kushkipagos.android.Kushki;
import com.kushkipagos.android.KushkiEnvironment;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.AskQuestionnaire;
import com.kushkipagos.android.SecureValidation;

abstract class AbstractRequestSecureValidationInfoAsyncTask extends AsyncTask<AskQuestionnaire, Void, SecureValidation> {

    protected final Kushki kushki;
    private final Context context;

    AbstractRequestSecureValidationInfoAsyncTask(Context context) {
        this.context = context;
        kushki = new Kushki("20000000107415376000", "COP", KushkiEnvironment.TESTING);
    }

    protected SecureValidation doInBackground(AskQuestionnaire... args) {
        try {
            return requestAskQuestionnaire(args[0]);
        } catch (KushkiException kushkiException) {
            throw new RuntimeException(kushkiException);
        }
    }

    protected void onPostExecute(SecureValidation secureValidation) {
        if (secureValidation.isSuccessful()) {
            showToast("QuestionnaireCode " + secureValidation.getQuestionnaireCode());
            System.out.println(secureValidation.getQuestions());
        } else {
            showToast("ERROR: " + secureValidation.getCode() + " " + secureValidation.getMessage());
        }
    }

    protected abstract SecureValidation requestAskQuestionnaire(AskQuestionnaire askQuestionnaire) throws KushkiException;

    private void showToast(String text) {
        Toast toast = Toast.makeText(context, text, Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: app/src/main/java/com/kushkipagos/kushkidemo/contracts/AbstractRequestSecureValidationInfoAsyncTask.kt
package com.kushkipagos.kushkidemo.contracts

import android.content.Context
import android.widget.Toast
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushki.Kushki
import com.kushkipagos.kushki.KushkiEnvironment
import com.kushkipagos.kushki.SecureValidation
import com.kushkipagos.kushkidemo.helpers.CoroutineTask
import com.kushkipagos.models.AskQuestionnaire

abstract class AbstractRequestSecureValidationInfoAsyncTask:
    CoroutineTask<AskQuestionnaire, Void, SecureValidation> {

    protected var _kushki: Kushki
    private var _context: Context

    constructor(context: Context){
        _context = context
        _kushki = Kushki("c308a8af32114b8c97f8ba191149df9b", "COP", KushkiEnvironment.QA)
    }

    @Throws(KushkiException::class)
    abstract fun requestAskQuestionnaire(askQuestionnaire: AskQuestionnaire) : SecureValidation

    override fun doInBackground(vararg params: AskQuestionnaire?): SecureValidation {
        try {
            return requestAskQuestionnaire(params[0]!!);
        } catch (kushkiException: KushkiException) {
            throw RuntimeException(kushkiException);
        }
    }

    override fun onPostExecute(param: SecureValidation?) {
        val secureValidation = param!!
        if (secureValidation.isSuccessful) {
            showToast("QuestionnaireCode: ${secureValidation.questionnaireCode}")
            println(secureValidation.questions)
        } else {
            showToast("ERROR: ${secureValidation.code} - ${secureValidation.message}")
        }
    }

    fun showToast(text: String){
        val toast = Toast.makeText(_context, text, Toast.LENGTH_LONG)
        toast.show()
    }

}